src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2647,7 +2646,7 @@ private void EnsureDescriptorsInitialized()
             {
                 // get the metadata via reflection.
                 Debug.Assert(m_rawManifest == null);
-                m_rawManifest = CreateManifestAndDescriptors(this.GetType(), Name, this);
+                m_rawManifest = CreateManifestAndDescriptors(typeof(EventSource), Name, this);
```

Comment:
This needs to walk the members on types deriving from EventSource - I assume this is why the tests are failing.

This line needs to be kept as-is and we need something like https://github.com/mono/linker/issues/1196 in the linker.

Reply:
Thanks for catching that! I reverted this for the time being since the issue you point is still open